### PR TITLE
feat: Fix Storefinder Radius and make configurable

### LINKED ISSUE #6275: Storefinder "Use my location" does not work - Radius is hardcoded
#5727  Environment Details

- Spartacus:  1.4
- Browser: Chrome
- OS: Mac

## Steps to Reproduce
Click the "Use my location"

https://spartacus.c39j2-walkersde1-d4-public.model-t.cc.commerce.ondemand.com/electronics-spa/en/USD/store-finder

## Observed Results
No results returned

## Expected Results
Results to show. 

## Repository Used
https://spartacus.c39j2-walkersde1-d4-public.model-t.cc.commerce.ondemand.com/electronics-spa/en/USD/store-finder

## Offer to Fix
No

## Additional Information

The issue is caused by the radius parameter. At the moment, this is hardcoded to use 10000000

According to the documentation -https://developers.google.com/places/web-service/search

The maximum radius should be  50 000 meters. Ideally, this should also be configurable.

### PR DESCRIPTION
**Pull request by Zeyber**
Closes: #6275

### CODE REVIEW

**Review by Pucek9 — COMMENTED** [review]
Hard to do local QA (You are using this API without a key. See http://developers.google.com/maps/documentation/javascript/error-messages?utm_source=maps_js&utm_medium=degraded&utm_campaign=billing#api-key-and-billing-errors)

**Review by Pucek9 — CHANGES_REQUESTED** [review]
Please use `yarn prettier:fix` because of problems here:
projects/core/src/store-finder/config/default-store-finder-config.ts
projects/core/src/store-finder/connectors/store-finder.adapter.ts
projects/core/src/store-finder/connectors/store-finder.connector.ts
projects/core/src/store-finder/store/effects/find-stores.effect.ts

**Review by Pucek9 — CHANGES_REQUESTED** [review]
-

**Comment by Zeyber** [comment]
> Hard to do local QA (You are using this API without a key. See http://developers.google.com/maps/documentation/javascript/error-messages?utm_source=maps_js&utm_medium=degraded&utm_campaign=billing#api-key-and-billing-errors)

The storefinder still works in dev mode without a key.

**Comment by Zeyber** [comment]
QA Steps for Config:
- Modify the `raidius` in `default-store-finder-config.ts`
- Navigate to storefinder
- Click `Use My Location`
- Check that the radius parameter in the request uses the modified value in the default config

image


Note: New radius may result in no stores found depending on your testing location because the old radius was large enough of 10000000 was large enough to cover the world.
You can change your location using dev tools to test in a location with stores (eg. Tokyo).

**Review by Pucek9 — DISMISSED** [review]

File: projects/storefrontlib/src/cms-components/storefinder/components/store-finder-search-result/store-finder-search-result.component.ts [hunk 1]
```diff
@@ -24,10 +25,12 @@ export class StoreFinderSearchResultComponent implements OnInit, OnDestroy {
   searchConfig: SearchConfig = {
     currentPage: 0,
   };
+  radius: number;
 
   constructor(
     private storeFinderService: StoreFinderService,
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/storefinder/components/store-finder-search-result/store-finder-search-result.component.ts:31]: Could we also in this PR make those dependencies protected?

**Review by marlass — CHANGES_REQUESTED** [review]
Could you also add StoreFinderConfig to StorefrontConfig and change all properties of googleMaps to optional? Then it would be following the same convention as any other configuration.

File: docs/migration/2_0.md [hunk 2]
```diff
@@ -358,3 +359,6 @@ This component was created with easy usage in mind. The only thing you have to d
 | `RegisterFormComponent` | method | - | `registerUser` |
 | `CustomFormValidators` | validator | `emailDomainValidator` | removed |
 | `CustomFormValidators` | validator | `matchPassword` | removed |
+
+## Store Finder Changes
+Paramater `radius` of `googleMaps?` parameter in `StoreFinderConfig` is now required.
```
Comment (marlass) [inline docs/migration/2_0.md]: I would change that. To only say that the radius is now configurable through the config. It is no longer required after the latest config changes ;)

**Review by marlass — CHANGES_REQUESTED** [review]

**Review by marlass — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**